### PR TITLE
Adding automatic npm-package deployment to travis

### PR DESCRIPTION
**Pull request by josojo**
With this PR, I am trying to introduce the same npm package deployment, as we currently have for dex-contracts.

### CODE REVIEW

**Review by bh2smith — COMMENTED** [review]
This is very exciting news! Let me know when its ready!

File: .travis.yml [hunk 1]
```diff
@@ -13,3 +13,25 @@ script:
...
+  - test "v$PACKAGE_VERSION" = "$TRAVIS_TAG"
+  - mkdir -p deploy
+  - yarn pack --filename deploy/dex-liquidity-provision-$PACKAGE_VERSION.tgz
+deploy:
+  provider: s3
+  bucket: $BUCKET_NAME
+  access_key_id: $AWS_ACCESS_KEY_ID
+  secret_access_key: $AWS_SECRET_ACCESS_KEY
+  local_dir: deploy
+  skip_cleanup: true
```
Comment (bh2smith) [inline .travis.yml:27]: If possible I would prefer not to skip cleanup.

Comment (josojo) [reply .travis.yml:27]: @nlordell Why is the clean_up skipped in dex-contracts?

Comment (josojo) [reply .travis.yml:28]: @nlordell Do you know why the clean up is skipped?

**Comment by josojo** [comment]
The npm package publication was successful: https://www.npmjs.com/package/@gnosis.pm/dex-liquidity-provision 

The next version will be published from master branch :) 
@fedgiac Can you review this PR?

**Comment by fedgiac** [comment]
Mmh, I'm a bit confused about the purpose of this PR. Who would be using this NPM package, and how can it be used? I don't see an entry point in `package.json`, so to my understanding it can't be imported directly in another script, each required file must be imported separately. I didn't see any change that would make the script work standalone with `npx` like it's the case for example with `eslint`. (But this would be an interesting concept!)

In any case, I'm not familiar with deploying to NPM, so I don't think I'm a good candidate to review this PR. Otherwise, I don't see any particular issue.

**Comment by josojo** [comment]
>Who would be using this NPM package,
cmm-app guys would want to run parts of our scripts. For example, if they want to do token-transfers, then they can just use our function:` buildTransferApproveDepositFromOrders`

This PR is really just about the npm package deployment, not about the package itself

File: .travis.yml (hunk 1 above)
Comment (fleupold) [reply .travis.yml:28]: https://docs.travis-ci.com/user/deployment-v2#cleaning-up-the-git-working-directory I assume we need the artefacts that were generated in the prepack step and are not part of the repo. If we cleaned up before deployment all those would be reset.

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2Fb5ZNou8CKvBCg%2Fgiphy.gif&token=lgtm" data-gitmeme-token="lgtm"></a>

**Pull request merged**